src/data/set/basic.lean

```diff
@@ -841,6 +843,16 @@ theorem forall_not_of_sep_empty {s : set α} {p : α → Prop} (H : {x ∈ s | p
 
 @[simp] lemma sep_univ {α} {p : α → Prop} : {a ∈ (univ : set α) | p a} = {a | p a} := univ_inter _
 
+@[simp] lemma sep_union : {x ∈ s ∪ t | p x} = {x ∈ s | p x} ∪ {x ∈ t | p x} :=
```

Comment:
Should these not be their own PR?

Reply:
I think that's small enough not to warrant another PR.

Reply:
Given this PR has sat around for a while due to disagreement elsewhere, I think puting these `sep` lemmas in a separate PR is probably a good idea.

Reply:
I'm not quite sure what you mean by disagreement elsewhere, but I do agree.

Reply:
I have prepared #16566 as suggested.